### PR TITLE
Added generic function for search by alt id.

### PR DESCRIPTION
**Pull request by sdas-infoblox**

### CODE REVIEW

File: object_manager.go [hunk 1]
```diff
@@ -301,3 +556,93 @@ func (objMgr *ObjectManager) UpdateZoneDelegated(ref string, delegate_to []NameS
...
+
+// validateObjByInternalId validates the object by comparing the given internal with the object's internal id
+func validateObjByInternalId(res interface{}, internalId, eaNameForInternalId string) (bool, error) {
+	var success bool
+	if res == nil {
+		return success, nil
+	} else if strings.TrimSpace(internalId) == "" {
+		// return object if internal id is empty
+		success = true
+		return success, nil
```
Comment (hemanthKa677) [inline object_manager.go:617]: why it is returned true here

**Review by hemanthKa677 — COMMENTED** [review]
besides that comment, looks good to me

File: object_manager.go (hunk 1 above)
Comment (sdas-infoblox) [reply object_manager.go:617]: We should return the return response received when fetched by ref and the internal id is empty. I should have updated the comment to `return true if response is not nil and internal id is empty`.

**Review by hemanthKa677 — APPROVED** [review]
LGTM

**Pull request merged**